File: data/EXP01/python/2410/2410.output_2.py
# @lc app=leetcode id=2410 lang=python3
class Solution:
    def matchPlayersAndTrainers(self, players: List[int], trainers: List[int]) -> int:
        players.sort()
        trainers.sort()
        
        count = 0
        player_idx = 0
        trainer_idx = 0
        
        while player_idx < len(players) and trainer_idx < len(trainers):
            if players[player_idx] <= trainers[trainer_idx]:
                count += 1
                player_idx += 1
                trainer_idx += 1
            else:
                trainer_idx += 1
        
        return count

File: data/EXP01/golang_no_system_var/2410/2410.output_2.go
func matchPlayersAndTrainers(players []int, trainers []int) int {
    sort.Ints(players)
    sort.Ints(trainers)

    count := 0

    for _, player := range players {
        found := false
        for i, trainer := range trainers {
            if player <= trainer {
                found = true
                count++
                trainers = append(trainers[:i], trainers[i+1:]...)
                break
            }
        }
        if !found {
            break
        }
    }

    return count
}
